docs/proposals/application/single_cluster_deployment_application.md

```diff
@@ -0,0 +1,68 @@
+---
+title: Single cluster deployment application
+authors:
+- @yeyeye2333
+reviewers:
+approvers:
+
+creation-date: 2024-07-24
+
+---
+
+## Single Cluster Deployment Application
+
+### Summary
+
+This proposal aims to modify the application API to allow setting the application's destination to a cluster, reducing the operational steps required for Kurator in a single-cluster environment and increasing Kurator's flexibility.
+
+### Motivation
+
+Kurator leverages Fleet for multi-cluster management. However, this has resulted in the current application deployment being tightly coupled with Fleet, even requiring Fleet creation for single-cluster usage. Therefore, we aim to decouple the single-cluster client from Fleet.
+
+#### Goals
+
+- Modify the application API to support setting the application's destination to a cluster.
+- Simplify the deployment process for single-cluster environments by eliminating the need for Fleet.
+
+### Proposal
+
+Currently, the application registration is handled in the fleet-manager. The proposal is to move the application registration to the cluster-operator to support setting the application's destination to a cluster.
+
+### Design Details
+
+We will delve into the API design required to support setting the application's destination to a cluster. The following are the specific changes to the application:
+
+```go
+// ApplicationDestination defines the configuration to dispatch an artifact to a fleet or specific clusters.
+// Fleet and cluster are mutually exclusive.
+type ApplicationDestination struct {
```

Comment:
change fleet-manager API or cluster-operator API ?

Reply:
The application is currently under fleet-manager, and I want to put it under cluster-operator

Reply:
application as a feature that depends on fleet, why put it in cluster-operator？
Even if you don't rely on fleet in a single cluster, fleet is still needed in a multi-cluster environment

Reply:
Deployment of application in single cluster without Fleet, does it mean that Fleet Manager does not need to be installed? So I need to register the application in cluster-operator. Will registering the application in cluster-operator affect fleet's use of the application?

Reply:
No. 
Deployment of application in single cluster without Fleet, means not need to create fleet.
Rather than do not install fleet manager.